static/js/country_details.js

```diff
@@ -0,0 +1,144 @@
+            }
+        });
+}
+
+const convertAbbrToCountryName = (abbr, data) => {
+    const convertedCountryNames = [];
+
+    abbr[0] === 'None' ? convertedCountryNames.push(abbr[0]) :
+        abbr.forEach((abbr) => {
+            const foundCountry = data.find(country => country["alpha3Code"] === abbr);
+
+            convertedCountryNames.push(foundCountry.name);
+        });
+
+    return convertedCountryNames;
+}
+
+const displayCard = (flag, name, nativeName, population, region, subregion, capital, topLevelDomain, currencies, languages, borders) => {
+    const card =
+        `<img class="detailed-flag" src="${flag}">
+    <div class="details-container">
+        <h2 class="detail-page-name">${name}</h2>
+
+        <ul class="details">
+            <li class="details-li"><strong>Native Name: </strong>${nativeName}</li>
+            <li class="details-li"><strong>Population: </strong>${population.toLocaleString('en-US')}</li>
+            <li class="details-li"><strong>Region: </strong>${region}</li>
+            <li class="details-li"><strong>Sub Region: </strong>${subregion}</li>
+            <li class="details-li"><strong>Capital: </strong>${capital}</li>
+        </ul>
+
+        <ul class="details">
+            <li class="details-li"><strong>Top Level Domain: </strong>${topLevelDomain}</li>
+            <li class="details-li"><strong>Currencies: </strong>${currencies.join(', ')}</li>
+            <li class="details-li"><strong>Languages: </strong>${languages.join(', ')}</li>
+        </ul>
+
+        <div class="border-countries-container">
+            <h3 class="border-countries-title"><strong>Border Countries:</strong></h3>
+            <div class="border-countries"></div>
+        </div>
+    </div>`;
+
+    return card
+}
+
+const displaySelectedCountry = () => {
+    const selectedCountry = localStorage.getItem('selected-country');
+
+    displayLoader(true)
+
+    fetchDetailedData(detailedInfo)
+        .then((data) => {
+            const countryToDisplay = data.find((country) => country.name === selectedCountry);
+            const currencies = [];
+            const languages = [];
+            let borderCountriesAbbr = [];
+
+            countryToDisplay.capital === undefined ? countryToDisplay.capital = 'None' : null;
+
+            countryToDisplay.currencies === undefined ?
+                currencies.push('None') :
+                countryToDisplay.currencies.forEach((currency) => { currencies.push(currency.name) });
+
+            countryToDisplay.languages.forEach((language) => { languages.push(language.name) });
+
+            countryToDisplay.borders === undefined ?
+                borderCountriesAbbr = ['None'] :
+                countryToDisplay.borders.forEach((country) => { borderCountriesAbbr.push(country) });
+
+            const borderCountries = convertAbbrToCountryName(borderCountriesAbbr, data);
+
+            detailedCountryCard.innerHTML = displayCard(
```

Comment:
Dit is wel netjes maar denk dat het beter zou zijn om ipv elke parameter zo los mee te sturen, om gwn `countryToDisplay` mee te sturen. ( zie line 52 )